### PR TITLE
always define a "next" map for a UMAPINFO entry

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
This fixes the desync in partest4.lmp. Works the same now as in PrBoom+ stable and master (checked).

**Comment by fabiangreffrath** [comment]
Will I also need to add the default endpic handling from here? 
https://github.com/coelckers/prboom-plus/blob/268931021606ed391966013b15456220c917a96e/prboom2/src/umapinfo.cpp#L685-L690

**Comment by fabiangreffrath** [comment]
> Will I also need to add the default endpic handling from here? 

I don't think that's necessary. I just renamed the map and UMAPINFO entry in partest.wad to MAP30, "finished" the map and was presented with the CAST ending. :+1:

File: src/u_mapinfo.c [hunk 1]
```diff
@@ -667,6 +668,15 @@ static int ParseStandardProperty(u_scanner_t* s, mapentry_t *mape)
     U_GetNextToken(s, true);
   } while (U_CheckToken(s, ','));
 
+  if (!mape->nextmap[0] && !mape->endpic[0])
+  {
+    int ep, map;
+
+    G_ValidateMapName(mape->mapname, &ep, &map);
+
+    strcpy(mape->nextmap, MAPNAME(ep, map + 1));
```
Comment (kraflab) [inline src/u_mapinfo.c:677]: This is the same logic as pr+, which is good, but looking at it I release this makes the default progression map 32 -> map 33 🤦 
Just confirmed in dsda-doom you crash when leaving map 32 if there's a map 32 entry without next defined...

Comment (kraflab) [reply src/u_mapinfo.c:677]: It's probably best to just leave it as is I guess. There isn't exactly a "normal" progression of maps in umapinfo, so maybe this is intentional.

**Review by kraflab — APPROVED** [review]

**Comment by fabiangreffrath** [comment]
> I don't think that's necessary. I just renamed the map and UMAPINFO entry in partest.wad to MAP30, "finished" the map and was presented with the CAST ending. +1

Hm, doing the same for E3M8 in Doom 1 leads to E3M9, though, instead of the bunny ending. Why does this happen, and does it need to be fixed?

**Comment by fabiangreffrath** [comment]
> Hm, doing the same for E3M8 in Doom 1 leads to E3M9, though, instead of the bunny ending. Why does this happen, and does it need to be fixed?

To answer my own question: For Doom 2, the `F_StartFinale()` call happens in `G_WorldDone()` and if there is a `gamemapinfo` entry available, but neither an `intertext` or an `endpic` field set, it goes the Vanilla code path. For Doom 1, the `F_StartFinale()` call happens in `G_DoCompleted()` only if `gameaction == ga_victory`, which in turn happens only if an `endpic` field is set. However, once a `gamemapinfo` entry is available, the entire Vanilla code is skipped (`goto frommapinfo`).

So, yes, in order to comply with PrBoom+'s default UMAPINFO behavior (and it's still the reference implementation), I'll have to apply a fix for this as well. Doing so, I have seen that I applied the previous patch at the wrong location in the code, but I'll fix this with the next PR.

**Pull request merged**